eclipse-collections/src/main/java/org/eclipse/collections/impl/lazy/CompositeIterable.java

```diff
@@ -122,6 +122,12 @@ public void add(Iterable<E> iterable)
     }
 
     @Override
+    public int size()
+    {
+       return (int) this.iterables.sumOfInt(iterable -> Iterate.sizeOf(iterable));
```

Comment:
Now that I see it written out, I realize that it can be shortened to `this.iterables.sumOfInt(Iterate::sizeOf)`. Either way (with or without this little refactoring) this looks good to go now.